### PR TITLE
TandemFoil: gc=1.5 transfer at lr=2e-4 (best config)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

gc=1.5 gave a 26% improvement on AirfRANS over gc=1.0 (PR #2737). TandemFoil has never tested gc=1.5 — its current best (PR #2610) uses the default grad-clip. Transferring gc=1.5 to TandemFoil at the winning lr=2e-4 + T_max=10 config may unlock a similar basin-depth improvement by allowing larger gradient signals to propagate while still controlling explosive peaks.

The mechanism: at lr=2e-4 with rapid T_max=10 cycling, LR peaks are relatively small. gc=1.5 (vs tighter values) allows the optimizer to take larger steps during these peaks, potentially reaching deeper loss basins. This is the same dynamic that produced the AirfRANS breakthrough.

## Instructions

Run exactly this command (180-min budget):

```bash
cd target/icml2026 && SENPAI_TIMEOUT_MINUTES=180 python train.py \
  --dataset tandemfoil --optimizer lion --lr 2e-4 \
  --cosine-t-max 10 --grad-clip 1.5 \
  --no-use-ema --model-slices 64 \
  --enable-fourier --enable-te-coord-frame \
  --enable-cp-panel --enable-cp-panel-tandem-only \
  --asinh-pressure --residual-prediction \
  --enable-pressure-prior-addition \
  --epochs 999 \
  --wandb_group tandem-lr2e4-gc
```

This is the PR #2610 winning config with only `--grad-clip 1.5` added. One hypothesis, one change.

Report `val_primary/surface_pressure_mae` at best epoch and final epoch.

## Baseline

- **Dataset:** TandemFoil
- **Primary metric:** `val_primary/surface_pressure_mae` (lower is better)
- **Current best:** 45.07 at epoch 107
- **Best PR:** #2610 (tetsuo — T_max=10, 3L/192d, Fourier + physics + no-EMA, slices=64, Lion lr=2e-4, 119 epochs, 180-min)
- **Baseline W&B run:** ixs1rqgk
- **Reproduce baseline:** `cd target/icml2026 && python train.py --dataset tandemfoil --optimizer lion --lr 2e-4 --cosine-t-max 10 --no-use-ema --model-slices 64 --enable-fourier --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --epochs 999`

**Target to beat:** `val_primary/surface_pressure_mae < 45.07`

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT alphonse:

## Results

### gc=1.5 transfer to TandemFoil at lr=2e-4 — **NEW BEST (-29.4%)**

| Metric | Baseline (#2610) | gc=1.5 (this PR) | Delta |
|--------|------------------|-------------------|-------|
| **val_primary/surface_pressure_mae** | 45.07 (ep 107) | **31.82 (ep 156)** | **-29.4%** |
| test_primary/surface_pressure_mae | 55.25 | **33.65** | **-39.1%** |

### Per-split breakdown (best epoch 156 val / test)

| Split | Val MAE | Test MAE |
|-------|---------|----------|
| single_in_dist | 34.89 | 30.68 |
| geom_camber_rc | 42.04 | 41.70 |
| geom_camber_cruise | 19.23 | 35.31 |
| re_rand | 31.14 | 26.91 |

### Training details

- **W&B run:** [93lxizt7](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/93lxizt7)
- **Epochs:** 162 (180-min timeout, ~1.11 min/epoch)
- **Best epoch:** 156 (val=31.82), final epoch 162 (val=33.31)
- **Grad clip behavior:** Heavy early clipping (747/750 steps at epoch 1) → minimal late (7-10/750 steps by epoch 160). Mean grad norm decreased from 9.74 → 0.37 over training.
- **Command:**
  ```bash
  cd target/icml2026 && SENPAI_TIMEOUT_MINUTES=180 python train.py \
    --dataset tandemfoil --optimizer lion --lr 2e-4 \
    --cosine-t-max 10 --grad-clip 1.5 \
    --no-use-ema --model-slices 64 \
    --enable-fourier --enable-te-coord-frame \
    --enable-cp-panel --enable-cp-panel-tandem-only \
    --asinh-pressure --residual-prediction \
    --enable-pressure-prior-addition \
    --epochs 999 \
    --wandb-group tandem-lr2e4-gc \
    --wandb-name "alphonse/tandem-gc15-lr2e4"
  ```

### What happened

gc=1.5 transfer from AirfRANS to TandemFoil is a **clear, decisive win**. The improvement (-29.4% val, -39.1% test) is even larger than the 26% gain gc=1.5 produced on AirfRANS in PR #2737.

The mechanism matches the hypothesis: at lr=2e-4 with T_max=10 rapid cycling, gc=1.5 allows larger gradient signals during LR peaks while still controlling explosive gradients. Early training shows heavy clipping (99.6% of steps clipped at epoch 1), which decays naturally as the model stabilizes (only ~1% clipped by epoch 160). This suggests gc=1.5 is most valuable during the initial high-gradient phase — it prevents catastrophic spikes while allowing faster basin exploration.

The training curve shows consistent downward envelope with T_max=10 oscillation. The model was still improving at timeout (loss 0.052 at ep 162 vs 0.061 at best ep 156) — more training would likely push lower.

Note: `geom_camber_cruise/mae_vol_p` shows Infinity in test, which appears to be an evaluation artifact (likely a single problematic sample).

### Suggested follow-ups

1. **gc=1.5 + 5L/256d architecture**: Combine gc=1.5 with the larger architecture from PR #2595 (which scored 52.81 at lr=3e-4). The gc=1.5 improvement should compound with architecture scaling.
2. **gc=2.0 sweep**: Test whether even looser clipping (gc=2.0) finds deeper basins at lr=2e-4 on TandemFoil, as the AirfRANS data suggests a non-monotonic relationship.
3. **Longer training at gc=1.5**: Model was still improving at 162 epochs — a 360-min budget could push significantly lower.
4. **gc=1.5 transfer to DrivAerML**: If gc=1.5 works on both AirfRANS and TandemFoil, it may be a universal improvement for all three datasets.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2753 because ADVISOR: Closing — your result of 31.82 does not beat the new 30.10 baseline. gc=1.5 appears too loose for TandemFoil. Thank you for testing this boundary, alphonse — it confirms tighter gradient clipping is needed for this dataset..
